airflow/operators/bash_operator.py

```diff
@@ -20,7 +20,7 @@
 import warnings
 
 # pylint: disable=unused-import
-from airflow.operators.bash import STDOUT, BashOperator, Popen, gettempdir  # noqa
+from airflow.operators.bash import BashOperator  # noqa
```

Comment:
Possibly a breaking change, but I'm leaning towards believing that no one used those objects...

Reply:
yes, totally possible.

we could let them remain and add a deprecation warning.

i am not sure --  is there a special syntax we have for "this must be removed at this release", like to automate deprecated removals?

whatever you suggest

Reply:
We usually create an issue with a milestone like `Airflow 2.1` and once releasing we are about to release we try to solve all of those. Am I correct @potiuk @kaxil or do we do something else?

Reply:
3.0 will be the next time we can remove anything (which will probably by 6-12 months timeframe -- likely 3.0 will be a "we've just removed deprecated things" rather than the huge release that 2.0 was)

Reply:
ok so sounds like what I need to do is put these imports back and add deprecation warning for 3.0 for them

Reply:
@ashb @turbaszek are you sure these imports really need to be preserved?

you could make the same argument about [these lines](https://github.com/apache/airflow/pull/13423/files#diff-912cc49726a411fd88f68b771c1de66996acd9b68c7e0e6b2e0caa7d8b23a188L21-L23).

or any time you remove an import from a module

users probably shouldn't be importing Popen from `bash_operator` anyway... WDYT?

if you think these left in, with deprecation warning, please also comment on the same issue for `airflow.operators.bash`

Reply:
This is probably one of the cases where removing this is _technically_ breaking, but people are unlikely to have used it, and it was never documented anyway, so is probably fine.

Probably.